project/component/src/events/details/details.js

```diff
@@ -0,0 +1,75 @@
+(function () {
+    const DetailsEvents = function (options) {
+        this.init = function (build) {
+            setEvents('new-evaluation', 'newEvaluation', build);
+            setEvents('evaluations', 'evaluations', build);
+            setEvents('logout', 'login', build);
+        };
+        this.destroy = function (build) {
+            removeEvents('new-evaluation', setEvents('new-evaluation', 'newEvaluation', build));
```

Comment:
I have this function removeEvents which has as a first parameter an id of a certain button, it takes the button and removes its event listener. The second parameter is the function for the event listener, so the same function for the removeEventListener. So what is does exactly is button.removeEventListener('click', function);

Reply:
Mkay, so let me see if I got this straight: `setEvents` _returns_ a reference to a function? And if so, is it _the same_ reference each and every time `setEvents` is invoked? Otherwise, what this is likely doing is trying to remove event listeners that _look_ the same, but are not referentially identical - much like calling person A, then trying to hang up on person B. :)

P.S.: Yeah, I know, I know, modern technology actually allows that to happen by putting calls on hold, but you get the point... :stuck_out_tongue:

Reply:
one word, four letters: bind  :)) a new function is created every time, so a new reference, even if it is the same function with the same code. That's why I should use bind to send the reference as a parameter. Something like that :D